ontotext-yasgui-web-component/src/components/share-saved-query-dialog/share-saved-query-dialog.tsx

```diff
@@ -0,0 +1,129 @@
+import {ServiceFactory} from "../../services/service-factory";
+import {TranslationService} from "../../services/translation.service";
+
+export type ShareSavedQueryDialogConfig = {
+  dialogTitle: string;
+  shareQueryLink: string;
+}
+
+@Component({
+  tag: 'share-saved-query-dialog',
+  styleUrl: 'share-saved-query-dialog.scss',
+  shadow: false,
+})
+export class ShareSavedQueryDialog {
+
+  private translationService: TranslationService;
+
+  @Element() hostElement: HTMLElement;
+
+  @Prop() config: ShareSavedQueryDialogConfig;
+
+  @Prop() serviceFactory: ServiceFactory
+
+  /**
+   * Event fired when the dialog is closed by triggering one of the close controls, e.g. close or
+   * cancel button as well as clicking outside of the dialog.
+   */
+  @Event() internalShareSavedQueryDialogClosedEvent: EventEmitter;
+
+  /**
+   * Internal event fired when saved query share link is copied in the clipboard.
+   */
+  @Event() internalSavedQueryShareLinkCopiedEvent: EventEmitter;
+
+  buildDialogConfig(): DialogConfig {
+    return {
+      dialogTitle: this.config.dialogTitle,
+      onClose: this.onClose.bind(this)
+    }
+  }
+
+  onCopy(evt: MouseEvent): void {
+    evt.stopPropagation();
+    this.copyToClipboard().then(() => {
+      this.internalSavedQueryShareLinkCopiedEvent.emit();
+    });
+  }
+
+  onClose(evt: MouseEvent): void {
+    const target = evt.target as HTMLElement;
+    evt.stopPropagation();
+    const isOverlay = target.classList.contains('dialog-overlay');
+    const isCloseButton = target.classList.contains('close-button');
+    const isCancelButton = target.classList.contains('cancel-button');
+    if (isOverlay || isCloseButton || isCancelButton) {
+      this.internalShareSavedQueryDialogClosedEvent.emit();
+    }
+  }
+
+  componentWillLoad(): void {
+    this.translationService = this.serviceFactory.get(TranslationService);
+  }
+
+  componentDidRender(): void {
+    const inputField: HTMLInputElement = document.querySelector('#shareLink');
+    inputField.focus();
+    // FIXME: For some reason this works only the first time dialog is opened
+    inputField.select();
+  }
+
+  private static fallbackCopyTextToClipboard(text: string): void {
+    const textArea = document.createElement("textarea");
+    textArea.value = text;
+
+    // Avoid scrolling to bottom
+    textArea.style.top = "0";
+    textArea.style.left = "0";
+    textArea.style.position = "fixed";
+
+    document.body.appendChild(textArea);
+    textArea.focus();
+    textArea.select();
+
+    try {
+      document.execCommand('copy');
+    } catch (err) {
+      console.error('Unable to copy', err);
+    }
+
+    document.body.removeChild(textArea);
+  }
+
+  private copyToClipboard(): Promise<void> {
+    if (!navigator.clipboard) {
+      ShareSavedQueryDialog.fallbackCopyTextToClipboard(this.config.shareQueryLink);
+      return Promise.resolve();
+    }
+    return navigator.clipboard.writeText(this.config.shareQueryLink).then(() => {
```

Comment:
We save here the passed value of the link, but user can change it in the input element. I think we have to mark the input as read only.